control-plane-operator/controllers/hostedcontrolplane/routecm/reconcile.go

```diff
@@ -0,0 +1,239 @@
+	"github.com/openshift/hypershift/control-plane-operator/controllers/hostedcontrolplane/manifests"
+	component "github.com/openshift/hypershift/support/controlplane-component"
+	"github.com/openshift/hypershift/support/util"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"k8s.io/utils/ptr"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	prometheusoperatorv1 "github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring/v1"
+)
+
+const (
+	ComponentName = "openshift-route-controller-manager"
+
+	ConfigMapName = "openshift-route-controller-manager-config"
+
+	configHashAnnotation       = "openshift-route-controller-manager.hypershift.openshift.io/config-hash"
+	servingPort          int32 = 8443
+)
+
+var _ component.DeploymentReconciler = &RouteControllerManagerReconciler{}
+
+type RouteControllerManagerReconciler struct {
+}
+
+func NewComponent() component.ControlPlaneComponent {
+	return &component.ControlPlaneWorkload{
+		DeploymentReconciler: &RouteControllerManagerReconciler{},
+		ResourcesReconcilers: []component.GenericReconciler{
+			component.NewReconcilerFor(&corev1.ConfigMap{}).
+				WithName(ConfigMapName).
+				WithReconcileFunction(ReconcileOpenShiftRouteControllerManagerConfig).
+				Build(),
+			component.NewReconcilerFor(&corev1.Service{}).
+				WithName(ComponentName).
+				WithReconcileFunction(ReconcileService).
+				WithPredicate(component.DisableIfAnnotationExist(hyperv1.DisableMonitoringServices)).
+				Build(),
+			component.NewReconcilerFor(&prometheusoperatorv1.ServiceMonitor{}).
+				WithName(ComponentName).
+				WithReconcileFunction(ReconcileServiceMonitor).
+				WithPredicate(component.DisableIfAnnotationExist(hyperv1.DisableMonitoringServices)).
+				Build(),
+		},
+
+		MultiZoneSpreadLabels: openShiftRouteControllerManagerLabels(),
+	}
+}
+
+// Name implements controlplanecomponent.DeploymentReconciler.
+func (r *RouteControllerManagerReconciler) Name() string {
+	return ComponentName
+}
+
+// ReconcileDeployment implements controlplanecomponent.DeploymentReconciler.
+func (r *RouteControllerManagerReconciler) ReconcileDeployment(cpContext component.ControlPlaneContext, deployment *appsv1.Deployment) error {
+	image := cpContext.ReleaseImageProvider.GetImage("route-controller-manager")
+
+	config := &corev1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      ConfigMapName,
+			Namespace: cpContext.HCP.Namespace,
+		},
+	}
+	if err := cpContext.Client.Get(cpContext, client.ObjectKeyFromObject(config), config); err != nil {
+		return fmt.Errorf("failed to get openshift controller manager config: %w", err)
+	}
+
+	configBytes, ok := config.Data[configKey]
+	if !ok {
+		return fmt.Errorf("openshift controller manager configuration is not expected to be empty")
+	}
+	configHash := util.ComputeHash(configBytes)
+
+	maxSurge := intstr.FromInt(0)
```

Comment:
I don't see the need to set this in every reconcile function. It should be done generically. If we have different kinds of deployment strategy, we could maybe just specify which kind we need.

Reply:
seems we already do that and stomp whatever is set here in the deploymentConfig if replicas > 1
https://github.com/openshift/hypershift/blob/f7fddd1961a24b40fa0a20d30489e9a2f7284a53/support/config/deployment.go#L88-L97

so just removing this code here would be enough, right?

Reply:
yes

Reply:
removed